Anlab.Mvc/Client/components/ClientId.tsx

```diff
@@ -0,0 +1,51 @@
+﻿
+import * as React from 'react';
+import Input from 'react-toolbox/lib/input';
+
+interface IClientIdProps {
+    clientId: string;
+    handleChange: Function;
+    isFromLab: boolean;
+}
+
+interface IClientIdInputState {
+    internalValue: string;
+    error: string;
+}
+
+export class ClientId extends React.Component<IClientIdProps, IClientIdInputState> {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            internalValue: this.props.clientId,
+            error: null
+        };
+    }
+    validate = (v: string) => {
+        let error = null;
+        if (v.trim() === "") {
```

Comment:
@laholstege this works but if v is null it would error (not sure that can happen).  Making the if statement `!v || v.trim() === ''` will make sure it isn't null or undefined.  If you just care about null or empty strings, if (!v) {} would work